BibBuddy/app/src/main/java/de/bibbuddy/BookNotesView.java

```diff
@@ -279,42 +283,64 @@ private void setupAddButton() {
     PopupMenu pm = new PopupMenu(getContext(), addButtonView);
     pm.getMenuInflater().inflate(R.menu.add_note_menu, pm.getMenu());
 
-    pm.setOnMenuItemClickListener(new PopupMenu.OnMenuItemClickListener() {
-      @Override
-      public boolean onMenuItemClick(MenuItem item) {
-        if (item.getItemId() == R.id.add_text_note) {
-          Bundle bundle = new Bundle();
-          bundle.putLong(LibraryKeys.BOOK_ID, bookId);
-          TextNoteEditorFragment nextFrag = new TextNoteEditorFragment();
-          nextFrag.setArguments(bundle);
-          getActivity().getSupportFragmentManager().beginTransaction()
-              .replace(R.id.fragment_container_view, nextFrag,
-                  LibraryKeys.FRAGMENT_TEXT_NOTE_EDITOR)
-              .addToBackStack(null)
-              .commit();
-
-        } /* else if (item.getItemId() == R.id.add_voice_note) {
-                    TODO: add features to add voice notes
-                }*/
-
-
-        return true;
+    pm.setOnMenuItemClickListener(item -> {
+      if (item.getItemId() == R.id.add_text_note) {
+        Bundle bundle = new Bundle();
+        bundle.putLong(LibraryKeys.BOOK_ID, bookId);
+        TextNoteEditorFragment textFrag = new TextNoteEditorFragment();
+        textFrag.setArguments(bundle);
+        requireActivity().getSupportFragmentManager().beginTransaction()
+            .replace(R.id.fragment_container_view, textFrag,
+                LibraryKeys.FRAGMENT_TEXT_NOTE_EDITOR)
+            .addToBackStack(null)
+            .commit();
+      } else {
+        checkRecordPermission();
       }
+
+      return true;
     });
 
+    addButtonView.setOnClickListener(v -> pm.show());
+  }
 
-    addButtonView.setOnClickListener(new View.OnClickListener() {
-      @Override
-      public void onClick(View v) {
-        pm.show();
-      }
-    });
+  private void checkRecordPermission() {
+    if (ContextCompat.checkSelfPermission(
+        context, Manifest.permission.RECORD_AUDIO) == PackageManager.PERMISSION_GRANTED) {
+      Bundle bundle = new Bundle();
+      bundle.putLong(LibraryKeys.BOOK_ID, bookId);
+      VoiceNoteEditorFragment voiceFrag = new VoiceNoteEditorFragment();
+      voiceFrag.setArguments(bundle);
+      requireActivity().getSupportFragmentManager().beginTransaction()
+          .replace(R.id.fragment_container_view, voiceFrag,
+              LibraryKeys.FRAGMENT_VOICE_NOTE_EDITOR)
+          .addToBackStack(null)
+          .commit();
+    } else if (shouldShowRequestPermissionRationale(Manifest.permission.RECORD_AUDIO)) {
+      showAudioRecordRequest();
+    } else {
+      requestPermissionLauncher.launch(
+          Manifest.permission.RECORD_AUDIO);
+    }
+  }
```

Comment:
A few newlines for better readability :)

```suggestion
  private void checkRecordPermission() {
    if (ContextCompat.checkSelfPermission(
        context, Manifest.permission.RECORD_AUDIO) == PackageManager.PERMISSION_GRANTED) {
        
      Bundle bundle = new Bundle();
      bundle.putLong(LibraryKeys.BOOK_ID, bookId);
      VoiceNoteEditorFragment voiceFrag = new VoiceNoteEditorFragment();
      voiceFrag.setArguments(bundle);
      
      requireActivity().getSupportFragmentManager().beginTransaction()
          .replace(R.id.fragment_container_view, voiceFrag,
              LibraryKeys.FRAGMENT_VOICE_NOTE_EDITOR)
          .addToBackStack(null)
          .commit();
          
    } else if (shouldShowRequestPermissionRationale(Manifest.permission.RECORD_AUDIO)) {
      showAudioRecordRequest();
      
    } else {
      requestPermissionLauncher.launch(
          Manifest.permission.RECORD_AUDIO);
    }
  }
```